lua/neo-tree/utils/init.lua

```diff
@@ -1009,6 +1009,53 @@ M.table_merge = function(base_table, override_table)
   return table_merge_internal(merged_table, override_table)
 end
 
+---Find the parent table to a target value inside a nested table structure.
+---
+---@param tbl table The table to search.
+---@param target_value any The value to search for.
+---@param max_depth number? Maximum depth to search (default: 5).
+---@param max_items number? Maximum number of items to check (default: 555).
+---@return table? parent_table The table containing the value, or nil if not found.
+M.table_find_by_value = function(tbl, target_value, max_depth, max_items)
```

Comment:
This was not needed. The config is already evaluated and passed to the command that initiates preview, and you should be able to pass that config down to where you need it. That pathway is important because it handles various overrides and checks for deprecation.

Aside for that: https://github.com/nvim-neo-tree/neo-tree.nvim/issues/1606